dipy/tracking/streamline.py

```diff
@@ -113,3 +116,169 @@ def select_random_set_of_streamlines(streamlines, select):
     len_s = len(streamlines)
     index = np.random.randint(0, len_s, min(select, len_s))
     return [streamlines[i] for i in index]
+
+
+def reduce_rois(rois, include):
```

Comment:
Definitely ``select_by_roi`` should be in streamline. Not sure about ``reduce_rois``. Feels like an utils... function.

Also I recommend to change the name to ``select_by_rois``. Should also this function replace Bago's ``target`` at some point?